### ISSUE TITLE
IndexError: index 4 is out of bounds for axis 0 with size 4 with pretrained polyphony_rnn

### ISSUE THREAD

**Issue by superMDguy** [tensorflow/magenta#532]
I've tried running the polyphony rnn using the docker container, anaconda, and the pip package, and every time I get the same error.  The command I'm using is `polyphony_rnn_generate --bundle_file=./polyphony_rnn.mag --output_dir=/magenta_data --num_outputs=10 --num_steps=128 --primer_pitches="[67,64,60]" --condition_on_primer=true `.   I have downloaded the pretrained polyphony_rnn.mag, and it's in the current directory.

Here's the full output of the command:
`INFO:tensorflow:Need to generate 124 more steps for this sequence, will try asking for 620 RNN steps
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArray is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArraySize is deprecated. It will cease to work in GraphDef version 16. Use TensorArraySizeV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayScatter is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayScatterV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayRead is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayReadV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayWrite is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayWriteV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayGather is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayGatherV2.
Traceback (most recent call last):
  File "/usr/local/bin/polyphony_rnn_generate", line 11, in <module>
    sys.exit(console_entry_point())
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/polyphony_rnn/polyphony_rnn_generate.py", line 278, in console_entry_point
    tf.app.run(main)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 43, in run
    sys.exit(main(sys.argv[:1] + flags_passthrough))
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/polyphony_rnn/polyphony_rnn_generate.py", line 274, in main
    run_with_flags(generator)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/polyphony_rnn/polyphony_rnn_generate.py", line 244, in run_with_flags
    generated_sequence = generator.generate(primer_sequence, generator_options)
  File "/usr/local/lib/python2.7/dist-packages/magenta/music/sequence_generator.py", line 204, in generate
    return self._generate(input_sequence, generator_options)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/polyphony_rnn/polyphony_sequence_generator.py", line 172, in _generate
    len(poly_seq) + rnn_steps_to_gen, poly_seq, **args)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/polyphony_rnn/polyphony_model.py", line 54, in generate_polyphonic_sequence
    modify_events_callback=modify_events_callback)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/shared/events_rnn_model.py", line 409, in _generate_events
    control_events, modify_events_callback)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/shared/events_rnn_model.py", line 343, in _beam_search
    final_state, temperature)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/shared/events_rnn_model.py", line 214, in _generate_branches
    all_event_sequences, all_inputs, all_final_state, temperature)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/shared/events_rnn_model.py", line 153, in _generate_step
    temperature)
  File "/usr/local/lib/python2.7/dist-packages/magenta/models/shared/events_rnn_model.py", line 111, in _generate_step_for_batch
    event_sequences, softmax)
  File "/usr/local/lib/python2.7/dist-packages/magenta/music/encoder_decoder.py", line 291, in evaluate_log_likelihood
    loglik += np.log(softmax[i][position][index])
IndexError: index 4 is out of bounds for axis 0 with size 4`

**Comment by Nimeas** [comment]
I have similar issue when trying to start interactive magenta_midi session with **polyphony_rnn** model.

Command: `magenta_midi   --input_port="Impulse:Impulse MIDI 1 20:0"   --output_port="FLUID Synth (4378):Synth input port (4378:0) 129:0"   --bundle_files=/tmp/polyphony_rnn.mag`

Output: 
```
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcurand.so locally
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_device.cc:885] Found device 0 with properties: 
name: GeForce GTX 950M
major: 5 minor: 0 memoryClockRate (GHz) 0.928
pciBusID 0000:01:00.0
Total memory: 3.95GiB
Free memory: 3.57GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:906] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 950M, pci bus id: 0000:01:00.0)
Loaded 'polyphony' generator bundle from file '/tmp/polyphony_rnn.mag'.

Instructions:
Start playing  when you want to begin the call phrase.
When you want to end the call phrase, stop playing and wait one clock tick.
Once the response completes, the interface will wait for you to begin playing again to start a new call phrase.

To end the interaction, press CTRL-C.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArray is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArraySize is deprecated. It will cease to work in GraphDef version 16. Use TensorArraySizeV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayScatter is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayScatterV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayRead is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayReadV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayWrite is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayWriteV2.
W tensorflow/core/framework/op_def_util.cc:332] Op TensorArrayGather is deprecated. It will cease to work in GraphDef version 16. Use TensorArrayGatherV2.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/interfaces/midi/midi_interaction.py", line 447, in run
    response_start_time + response_duration)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/interfaces/midi/midi_interaction.py", line 338, in _generate
    adjust_sequence_times(input_sequence, -zero_time), generator_options)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/music/sequence_generator.py", line 204, in generate
    return self._generate(input_sequence, generator_options)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/models/polyphony_rnn/polyphony_sequence_generator.py", line 172, in _generate
    len(poly_seq) + rnn_steps_to_gen, poly_seq, **args)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/models/polyphony_rnn/polyphony_model.py", line 54, in generate_polyphonic_sequence
    modify_events_callback=modify_events_callback)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/models/shared/events_rnn_model.py", line 409, in _generate_events
    control_events, modify_events_callback)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/models/shared/events_rnn_model.py", line 343, in _beam_search
    final_state, temperature)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/models/shared/events_rnn_model.py", line 214, in _generate_branches
    all_event_sequences, all_inputs, all_final_state, temperature)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/models/shared/events_rnn_model.py", line 153, in _generate_step
    temperature)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/models/shared/events_rnn_model.py", line 111, in _generate_step_for_batch
    event_sequences, softmax)
  File "/home/lukas/libs/miniconda2/envs/magenta/lib/python2.7/site-packages/magenta/music/encoder_decoder.py", line 291, in evaluate_log_likelihood
    loglik += np.log(softmax[i][position][index])
IndexError: index 4 is out of bounds for axis 0 with size 4
```

**Comment by cghawthorne** [comment]
This should be fixed by #565. You can try it out in our development environment, or we'll cut a new pip package soon.
